client/src/pages/sniffers/AddSnifferPage.tsx

```diff
@@ -124,17 +121,37 @@ function SimpleDomainComponent(props: {
   domain: string;
   onDomainChange: (domain: string) => void;
 }) {
+  const [error, setError] = React.useState<string | null>(null);
+  const DEBOUNCE_TIME = 1000;
+
+  const handleDomainChange = (newValue: string) => {
+    props.onDomainChange(newValue);
+    debouncedSearch(newValue);
+  };
+
+  const debouncedSearch = React.useRef(
```

Comment:
Why did you use useRef here?

Reply:
I wanted to make sure that this function will not change across re renders (it's causing the error message to flicker). 
But that you mentioned it, no point in using `useRef`, I changed this into `useCallback` as I think it's a more appropriate way to handle this. 

Do you prefer a different solution?

Reply:
Great!